include/stl2/view/drop.hpp

```diff
@@ -0,0 +1,117 @@
+// cmcstl2 - A concept-enabled C++ standard library
+//
+//  Copyright Casey Carter
+//  Copyright Eric Niebler
+//  Copyright Christopher Di Bella
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/caseycarter/cmcstl2
+//
+#ifndef STL2_VIEW_DROP_HPP
+#define STL2_VIEW_DROP_HPP
+
+#include <stl2/detail/fwd.hpp>
+#include <stl2/detail/meta.hpp>
+#include <stl2/detail/iterator/concepts.hpp>
+#include <stl2/detail/non_propagating_cache.hpp>
+#include <stl2/detail/range/access.hpp>
+#include <stl2/detail/range/concepts.hpp>
+#include <stl2/detail/view/view_closure.hpp>
+#include <stl2/view/all.hpp>
+#include <stl2/view/view_interface.hpp>
+
+STL2_OPEN_NAMESPACE {
+	namespace ext {
+		template <View R>
+		class drop_view : public view_interface<drop_view<R>> {
+			using D = iter_difference_t<iterator_t<R>>;
+		public:
+			drop_view() = default;
+
+			constexpr drop_view(R base, D count)
+				: base_(std::move(base)),
+				  count_(count)
+			{}
+
+			template <ViewableRange O>
+			requires
+				_ConstructibleFromRange<R, O>
+			constexpr drop_view(O&& o, D count)
+				: base_(view::all(std::forward<O>(o))),
+				  count_(count)
+			{}
+
+			constexpr R base() const { return base_; }
+
+			constexpr auto begin() requires !SimpleView<R> { return begin_impl(*this); }
+			constexpr auto begin() const requires Range<R const>
+			{ return begin_impl(*this); }
+
+			constexpr auto end() requires !SimpleView<R> { return end_impl(*this); }
+			constexpr auto end() const requires Range<R const>
+			{ return end_impl(*this); }
+
+			constexpr auto size() requires !SimpleView<R> && SizedRange<R> { return size_impl(*this); }
+			constexpr auto size() const requires SizedRange<R const> { return size_impl(*this); }
+		private:
+			R base_;
+			D count_;
+			mutable detail::non_propagating_cache<iterator_t<R>> begin_;
+
+			template <class X>
+			static constexpr auto begin_impl(X& x)
+			{
+				if (!x.begin_) {
+					if constexpr (SizedRange<__maybe_const<is_const_v<X>, R>>) {
+						D const dist = __stl2::distance(x.base_);
+						STL2_EXPECT(x.count_ >= 0);
+						x.begin_ = __stl2::next(__stl2::begin(x.base_), dist < x.count_ ? dist : x.count_);
+					}
+					else {
+						x.begin_ = __stl2::next(__stl2::begin(x.base_), x.count_);
```

Comment:
This should call the "bounded" `next` overload `next(begin(base_), count_, end(base_))` to avoid running past the end of the base range.